xbmc/Application.cpp

```diff
@@ -3884,6 +3884,10 @@ bool CApplication::OnMessage(CGUIMessage& message)
         // show info dialog about moved configuration files if needed
         ShowAppMigrationMessage();
 
+        // offer enabling addons at kodi startup that are disabled due to
+        // e.g. os package manager installation on linux
+        ConfigureAndEnableAddons();
```

Comment:
Q1: How does this effect Kodi startup time? Is it guaranteed that it will always be quick if no disabled  addons are found?

Q2: And how does Kodi "look" at this stage? Will there be "nothing" behind the dialog or the full Kodi GUI? How will it appear to the user?

Reply:
if there are no disabled add-ons we end up with an empty vector at the beginning of the function.
https://github.com/xbmc/xbmc/blob/b9ae9de3497c7f4025ecc056de9bb34352c7bd95/xbmc/Application.cpp#L4173
so i don't expect hard delay in that case.